internal/testrunner/runners/system/runner.go

```diff
@@ -395,12 +403,35 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 	// Reload test config with ctx variable substitution.
 	config, err = newConfig(config.Path, ctxt, serviceOptions.Variant)
 	if err != nil {
-		return result.WithError(errors.Wrap(err, "unable to reload system test case configuration"))
+		return result.WithError(fmt.Errorf("unable to reload system test case configuration: %w", err))
 	}
 
 	kib, err := kibana.NewClient()
 	if err != nil {
-		return result.WithError(errors.Wrap(err, "can't create Kibana client"))
+		return result.WithError(fmt.Errorf("can't create Kibana client: %w", err))
+	}
+
+	// Install the package before creating the policy, so we control exactly what is being
+	// installed.
+	logger.Debug("Installing package...")
+	installer, err := installer.NewForPackage(installer.Options{
+		Kibana:         kib,
+		RootPath:       r.options.PackageRootPath,
+		SkipValidation: true,
+	})
+	if err != nil {
+		return result.WithError(fmt.Errorf("failed to initialize package installer: %v", err))
+	}
+	_, err = installer.Install()
+	if err != nil {
+		return result.WithError(fmt.Errorf("failed to install package: %v", err))
+	}
+	r.deletePackageHandler = func() error {
```

Comment:
I couldn't just use `defer` here, I needed to use a handler to play well with the tear downs here.